README.md

```diff
@@ -11,6 +11,37 @@ To compile
 sbt tut
 ```
 
+## Style guide
+
+Use **\`\`\`tut:silent\`\`\`** for code that is compilable, use **\`\`\`scala\`\`\`** otherwise.
+
+Use a line of `.............` to separate between code examples. 
+
+
+There are multiple ways to translate haskell code into scala. For scala syntax, generally speaking we follow the common styles in FP scala codebase (e.g. typelevel/cats)
+
+### functions vs. methods
+
+If it's a "standalone" function, use the function syntax, i.e.
+```scala
+val f: A => B = ???
+```
+If it is inside a trait or class, use the method, i.e.
+
+```scala
+def f(a: A) : B = ???
+```
+
+### Currying
+
+Follow the common strategies in FP codebases, roughly
+
+* if there is one function parameter, create a dedicated parameter list for it and make it the last parameter list. 
+* if there are multple function parameters, group them togeher, and put them in a dedicated last paramater list. 
+* if the haskell code acutally use the curry application of the function, we curry the scala version to the point to that such application is mirrored in the scala code. 
```

Comment:
`s/acutally/actually`. Not super clear what you mean by this point. All Haskell code is using currying. Do you mean cases in which a function is "partially applied"? As for the scala version, do we partially apply with underscore? (as opposed to making the original function curried).
Also probably `s/to the point to that/so that`?

Reply:
I changed it to
```md
if the haskell code partially applied the function, partially apply with underscore in scala code. 
```